patch/0001-armhf-mandatory-configs-for-boot.patch

```diff
@@ -0,0 +1,201 @@
+-CONFIG_MTD_AR7_PARTS=m
++CONFIG_MTD_OF_PARTS=y
++CONFIG_MTD_AR7_PARTS=y
+ 
+ #
+ # User Modules And Translation Layers
+ #
+-CONFIG_MTD_BLKDEVS=m
+-CONFIG_MTD_BLOCK=m
++CONFIG_MTD_BLKDEVS=y
++CONFIG_MTD_BLOCK=y
+ CONFIG_MTD_BLOCK_RO=m
+ # CONFIG_FTL is not set
+ # CONFIG_NFTL is not set
+@@ -1817,8 +1817,10 @@ CONFIG_MTD_SWAP=m
+ #
+ # RAM/ROM/Flash chip drivers
+ #
+-# CONFIG_MTD_CFI is not set
++CONFIG_MTD_CFI=y
+ # CONFIG_MTD_JEDECPROBE is not set
++CONFIG_MTD_GEN_PROBE=y
++# CONFIG_MTD_CFI_ADV_OPTIONS is not set
+ CONFIG_MTD_MAP_BANK_WIDTH_1=y
+ CONFIG_MTD_MAP_BANK_WIDTH_2=y
+ CONFIG_MTD_MAP_BANK_WIDTH_4=y
+@@ -1829,6 +1831,10 @@ CONFIG_MTD_CFI_I1=y
+ CONFIG_MTD_CFI_I2=y
+ # CONFIG_MTD_CFI_I4 is not set
+ # CONFIG_MTD_CFI_I8 is not set
++CONFIG_MTD_CFI_INTELEXT=y
++CONFIG_MTD_CFI_AMDSTD=y
++CONFIG_MTD_CFI_STAA=y
++CONFIG_MTD_CFI_UTIL=y
+ CONFIG_MTD_RAM=m
+ # CONFIG_MTD_ROM is not set
+ # CONFIG_MTD_ABSENT is not set
+@@ -1850,8 +1856,8 @@ CONFIG_MTD_PLATRAM=m
+ CONFIG_MTD_DATAFLASH=m
+ # CONFIG_MTD_DATAFLASH_WRITE_VERIFY is not set
+ # CONFIG_MTD_DATAFLASH_OTP is not set
+-CONFIG_MTD_M25P80=m
+-CONFIG_MTD_SST25L=m
++CONFIG_MTD_M25P80=y
++CONFIG_MTD_SST25L=y
+ # CONFIG_MTD_SLRAM is not set
+ # CONFIG_MTD_PHRAM is not set
+ # CONFIG_MTD_MTDRAM is not set
+@@ -1878,7 +1884,7 @@ CONFIG_MTD_NAND_RICOH=m
+ # CONFIG_MTD_NAND_DISKONCHIP is not set
+ # CONFIG_MTD_NAND_DOCG4 is not set
+ CONFIG_MTD_NAND_CAFE=m
+-CONFIG_MTD_NAND_PXA3xx=m
++CONFIG_MTD_NAND_PXA3xx=y
+ CONFIG_MTD_NAND_NANDSIM=m
+ CONFIG_MTD_NAND_GPMI_NAND=m
+ # CONFIG_MTD_NAND_BRCMNAND is not set
+@@ -1902,12 +1908,12 @@ CONFIG_MTD_ONENAND_2X_PROGRAM=y
+ CONFIG_MTD_LPDDR=m
+ CONFIG_MTD_QINFO_PROBE=m
+ # CONFIG_MTD_LPDDR2_NVM is not set
+-CONFIG_MTD_SPI_NOR=m
++CONFIG_MTD_SPI_NOR=y
+ # CONFIG_MTD_MT81xx_NOR is not set
+ CONFIG_MTD_SPI_NOR_USE_4K_SECTORS=y
+ # CONFIG_SPI_CADENCE_QUADSPI is not set
+ # CONFIG_SPI_FSL_QUADSPI is not set
+-CONFIG_MTD_UBI=m
++CONFIG_MTD_UBI=y
+ CONFIG_MTD_UBI_WL_THRESHOLD=4096
+ CONFIG_MTD_UBI_BEB_LIMIT=20
+ # CONFIG_MTD_UBI_FASTMAP is not set
+@@ -3266,13 +3272,14 @@ CONFIG_VT_CONSOLE_SLEEP=y
+ CONFIG_HW_CONSOLE=y
+ CONFIG_VT_HW_CONSOLE_BINDING=y
+ CONFIG_UNIX98_PTYS=y
+-# CONFIG_LEGACY_PTYS is not set
++CONFIG_LEGACY_PTYS=y
++CONFIG_LEGACY_PTY_COUNT=256
+ # CONFIG_SERIAL_NONSTANDARD is not set
+ CONFIG_NOZOMI=m
+ CONFIG_N_GSM=m
+ # CONFIG_TRACE_SINK is not set
+ CONFIG_DEVMEM=y
+-# CONFIG_DEVKMEM is not set
++CONFIG_DEVKMEM=y
+ 
+ #
+ # Serial drivers
+@@ -3824,7 +3831,7 @@ CONFIG_SENSORS_MAX1668=m
+ CONFIG_SENSORS_MAX6639=m
+ CONFIG_SENSORS_MAX6642=m
+ CONFIG_SENSORS_MAX6650=m
+-# CONFIG_SENSORS_MAX6620 is not set
++CONFIG_SENSORS_MAX6620=m
+ # CONFIG_SENSORS_MAX6697 is not set
+ # CONFIG_SENSORS_MAX31790 is not set
+ # CONFIG_SENSORS_MCP3021 is not set
+@@ -7215,14 +7222,14 @@ CONFIG_JFFS2_RTIME=y
+ CONFIG_JFFS2_CMODE_PRIORITY=y
+ # CONFIG_JFFS2_CMODE_SIZE is not set
+ # CONFIG_JFFS2_CMODE_FAVOURLZO is not set
+-CONFIG_UBIFS_FS=m
++CONFIG_UBIFS_FS=y
```

Comment:
>CONFIG_UBIFS_FS [](start = 1, length = 15)

why y not m?

Reply:
Same as https://github.com/Azure/sonic-linux-kernel/pull/102#discussion_r339897354

Reply:
Done